src/main/java/org/entando/kubernetes/model/bundle/usage/ComponentUsage.java

```diff
@@ -0,0 +1,95 @@
+package org.entando.kubernetes.model.bundle.usage;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.entando.kubernetes.model.entandocore.EntandoCoreComponentUsage;
+import org.entando.kubernetes.model.entandocore.EntandoCoreComponentUsage.EntandoCoreComponentReference;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+@Builder
+public class ComponentUsage {
+
+    private String type;
+    private String code;
+    private boolean exist;
+    private int usage;
+    private boolean hasExternal;
+    private List<ComponentReference> references;
+
+
+    public boolean getHasExternal() {
+        return hasExternal;
+    }
+
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @Data
+    @Builder
+    public static class ComponentReference {
+
+        private String componentType;
+        private ComponentReferenceType referenceType;
+        private String code;
+        @JsonInclude(Include.NON_NULL)
+        private Boolean online;
+
+
+        /**
+         * Utility method to copy the fields present in the EntandoCoreComponentReference within the ComponentReference
+         * (used to contain and enrich the information present in the object returned by Entando Core).
+         * ComponentReference specific fields (for which there is no correspondence in the object returned by Entando
+         * Core) are not set as this is left to the business logic applied by the appropriate services.
+         *
+         * @param coreComponentReference portion of the response obtained from Entando Core concerning references for a
+         *                               specific component
+         * @return an object of type ComponentReference in which only the relative fields obtained from
+         *     EntandoCoreComponentReference are set
+         */
+        public static ComponentReference fromEntandoCore(EntandoCoreComponentReference coreComponentReference) {
+            return ComponentReference.builder().componentType(coreComponentReference.getType())
```

Comment:
Please if coreComponentReference is null return immediately to avoid NPE

Reply:
a NPE is not raised in the logic written (2 filters against NP are setted). Anyway this method could be used in other portions of the code, so your suggestion is right and wellcomed! Thank you